src/go/configgenerator/route_generator.go

```diff
@@ -144,6 +147,63 @@ func MakeRouteConfig(serviceInfo *configinfo.ServiceInfo) (*routepb.RouteConfigu
 	}, nil
 }
 
+func MakePathRewriteConfig(method *configinfo.MethodInfo, httpRule *commonpb.Pattern) *prpb.PerRouteFilterConfig {
+	if method.BackendInfo == nil {
+		return nil
+	}
+
+	if method.BackendInfo.TranslationType == confpb.BackendRule_APPEND_PATH_TO_ADDRESS {
+		if method.BackendInfo.Path != "" {
+			return &prpb.PerRouteFilterConfig{
+				PathTranslationSpecifier: &prpb.PerRouteFilterConfig_PathPrefix{
+					PathPrefix: method.BackendInfo.Path,
+				},
+			}
+		}
+	}
+	if method.BackendInfo.TranslationType == confpb.BackendRule_CONSTANT_ADDRESS {
+		constPath := &prpb.ConstantPath{
+			Path: method.BackendInfo.Path,
+		}
+		if strings.ContainsRune(httpRule.UriTemplate, '{') {
```

Comment:
This check can be done based on the new-added `UrlTemplate` parsing, by if the`variables` size > 0.

Reply:
+1, recommend checking the parsed struct instead of individual characters

Reply:
Maybe you can add a TODO here. I can replace later when applying `UrlTemplate`.

Reply:
Ack.  UrlTemplate parsing code is in, but not hooked up yet.  I will just add a /TODO here